src/app.jsx

```diff
@@ -364,11 +371,17 @@ var LoadForm = React.createClass({
     },
     render: function () {
         var releases = this.props.releases.map(function (val) {
-            return <option key={val}>{val}</option>;
+            return <option value={val}>{val}</option>;
         });
         var roles = this.props.roles.map(function (val) {
-            return <option key={val}>{val}</option>;
+            return <option value={val}>{val}</option>;
         });
+        var component = (this.props.params['component']) ? this.props.params['component']:"";
+        $("#component").attr("value", component);
+        var release = (this.props.params['release']) ? this.props.params['release']:"all";
+        $("#release option[value='" + release + "']").attr("selected", "selected");
+        var role = (this.props.params['role']) ? this.props.params['role']:"all";
+        $("#role option[value='" + role + "']").attr('selected','selected');
```

Comment:
component is a bit different with the other two.
the other two are abstracted in a common method.